app/models/doctor.rb

```diff
@@ -1,6 +1,7 @@
 class Doctor < ApplicationRecord
   has_many :reservations
   belongs_to :user
+  has_many :users, through: :reservations
```

Comment:
Would you mind fixing the two lines of code, because they seem to be confused, how can `doctor` belong to `user` in line 3, and then `doctor` has again many users in line 4?
I would suggest deleting line 4.

Reply:
Hey @moise-mulungu , I totally understand why you find it confusing. It may be confusing but it works very nicely. Let me show you the benefit of that relation in our case. 
In our case, we are creating a relation between a doctor and his patients through the reservation. With our association, we can do something like this : `doctor.users` to retrieve users or patients the doctor had reservations with. And in the other hand, we can do `doctor.user` to retrieve the user that created that doctor.